packages/@expo/cli/src/lint/lintAsync.ts

```diff
@@ -0,0 +1,111 @@
+import JsonFile, { type JSONObject } from '@expo/json-file';
+import * as PackageManager from '@expo/package-manager';
+import spawnAsync from '@expo/spawn-async';
+import fs from 'fs/promises';
+import path from 'path';
+
+import { Log } from '../log';
+import { selectAsync } from '../utils/prompts';
+
+const WITH_PRETTIER = `module.exports = {
+  root: true,
+  extends: ["expo", "prettier"],
+  plugins: ["expo", "prettier"],
+  rules: {
+    "prettier/prettier": ["warn"],
+  },
+};
+
+`;
+
+const ESLINT_ONLY = `module.exports = {
+  root: true,
+  extends: ["expo"],
+};
+`;
+
+const setupLinting = async (projectRoot: string) => {
+  const result = await selectAsync(
+    'No eslint config found. Would you like to set up linting for this project?',
+    [
+      {
+        title: 'Yes, eslint only',
+        value: 'eslint',
+      },
+      {
+        title: 'Yes, eslint and prettier',
+        value: 'eslint-and-prettier',
+      },
+      {
+        title: 'No',
+        value: 'no',
+      },
+    ]
+  );
+
+  if (result === 'no') {
+    return;
+  }
+
+  const packages = [
+    'eslint',
+    // 'eslint-config-expo',
+  ];
+
+  if (result === 'eslint-and-prettier') {
+    packages.push('prettier');
+    packages.push('eslint-config-prettier');
+    packages.push('eslint-plugin-prettier');
+  }
+
+  const manager = PackageManager.createForProject(projectRoot);
```

Comment:
The majority of this logic looks like it could fit in a prerequisite https://github.com/expo/expo/blob/cfb6721824c442691ed13c93ef4577ed8a62e59b/packages/%40expo/cli/src/start/doctor/typescript/TypeScriptProjectPrerequisite.ts#L1

Should also used versioned package installing